### PR TITLE
[14.0][MIG] stock_vertical_lift_server_env

### PR DESCRIPTION
**Pull request by nguyenhk99**
Dependency: https://github.com/OCA/stock-logistics-warehouse/pull/1090

### CODE REVIEW

**Review by simahawk — COMMENTED** [review]
LG, waiting for the dep to get merged

**Review by grindtildeath — APPROVED** [review]
LGTM

Can you please remove the test dependency now that #1090 is merged?

**Comment by simahawk** [comment]
runbot is :red_circle: because of this:

```
2021-09-21 03:48:11,106 174 INFO openerp_test odoo.modules.loading: loading stock_location_tray/demo/stock_location_tray_type_demo.xml
2021-09-21 03:48:11,127 174 WARNING openerp_test odoo.modules.loading: Module stock_location_tray demo data failed to install, installed without demo data
Traceback (most recent call last):
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 677, in _tag_root
f(rec)
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 580, in _tag_record
record = model._load_records([data], self.mode == 'update')
File "/home/odoo/odoo-14.0/odoo/models.py", line 4202, in _load_records
data['record']._load_records_write(data['values'])
File "/home/odoo/odoo-14.0/odoo/models.py", line 4139, in _load_records_write
self.write(values)
File "/home/odoo/odoo-14.0/odoo/models.py", line 3687, in write
real_recs._validate_fields(vals, inverse_fields)
File "/home/odoo/odoo-14.0/odoo/models.py", line 1266, in _validate_fields
check(self)
File "/home/odoo/build/OCA/stock-logistics-warehouse/setup/stock_location_tray/odoo/addons/stock_location_tray/models/stock_location_tray_type.py", line 105, in _location_check_rows_cols
).format(record.name, "\n".join(location_bullets))
odoo.exceptions.ValidationError: The tray type Small 8x is used by the following locations, it's size cannot be changed:

- WH/Stock/Tray
- WH/Stock/Vertical Lift/Shuttle 1/Tray 1A
- WH/Stock/Vertical Lift/Shuttle 2/Tray 2D

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/home/odoo/odoo-14.0/odoo/modules/loading.py", line 86, in load_demo
load_data(cr, idref, mode, kind='demo', package=package)
File "/home/odoo/odoo-14.0/odoo/modules/loading.py", line 69, in load_data
tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind)
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 733, in convert_file
convert_xml_import(cr, module, fp, idref, mode, noupdate)
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 799, in convert_xml_import
obj.parse(doc.getroot())
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 719, in parse
self._tag_root(de)
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 685, in _tag_root
)) from e
odoo.tools.convert.ParseError: while parsing /home/odoo/build/OCA/stock-logistics-warehouse/setup/stock_location_tray/odoo/addons/stock_location_tray/demo/stock_location_tray_type_demo.xml:15, near
<record model="stock.location.tray.type" id="stock_location_tray_type_small_8x">
<field name="name">Small 8x</field>
<field name="code">B20402</field>
<field name="rows">2</field>
<field name="cols">4</field>
</record>

```
could you have a look?
Seems related to the base PR tho.

File: stock_vertical_lift_server_env/models/vertical_lift_shuttle.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            "hardware": {"compute_default": "_compute_default_hardware"},
+            "server": {},
+            "port": {},
+            "use_tls": {},
+        }
+        sftp_fields.update(base_fields)
+        return sftp_fields
+
+    def _compute_default_hardware(self):
+        self.hardware = "simulation"
```
Comment (simahawk) [inline stock_vertical_lift_server_env/models/vertical_lift_shuttle.py]: does this work with multiple records? I'd use `self.update({"hardware": "simulation"})`

Comment (nguyenhk99) [reply stock_vertical_lift_server_env/models/vertical_lift_shuttle.py]: Thanks for your comment. i will work on it.

**Comment by nguyenhk99** [comment]
> LGTM
> 
> Can you please remove the test dependency now that #1090 is merged?

Sorry for the late reply. I updated the PR

**Comment by nguyenhk99** [comment]
I fixed runbot errors with the command git rebase and `self.update({"hardware": "simulation"})`. Thanks for everyone's help.

**Review by simahawk — APPROVED** [review]

**Pull request merged**